### PR TITLE
Add option to highlight self in kill messages

### PR DESCRIPTION
**Pull request by Azorlogh**
Motivation: #4186

List of changes:
- Add options clKillMessageHighlightSelf, clKillMessageNormalColor, clKillMessageHighlightColor
- Rework HUD panel in settings

hud_0

hud_1

in_game

## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [x] Tested in combination with possibly related configuration options
- [x] Written a unit test if it works standalone, system.c especially (not standalone)
- [x] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by Jupeyy** [comment]
Is `m_pLocalInfo` safe to use like this?
When I search it, every component NULL checks it

**Comment by Azorlogh** [comment]
Indeed you're right. I switched to using m_LocalClientID instead which worse case scenario will be -1.

File: src/game/variables.h [hunk 1]
```diff
@@ -49,6 +49,7 @@ MACRO_CONFIG_INT(ClShowChat, cl_showchat, 1, 0, 2, CFGFLAG_CLIENT | CFGFLAG_SAVE
 MACRO_CONFIG_INT(ClShowChatFriends, cl_show_chat_friends, 0, 0, 1, CFGFLAG_CLIENT | CFGFLAG_SAVE, "Show only chat messages from friends")
 MACRO_CONFIG_INT(ClShowChatSystem, cl_show_chat_system, 1, 0, 1, CFGFLAG_CLIENT | CFGFLAG_SAVE, "Show chat messages from the server")
 MACRO_CONFIG_INT(ClShowKillMessages, cl_showkillmessages, 1, 0, 1, CFGFLAG_CLIENT | CFGFLAG_SAVE, "Show kill messages")
+MACRO_CONFIG_INT(ClKillMessageHighlightSelf, cl_killmessage_highlightself, 0, 0, 1, CFGFLAG_CLIENT | CFGFLAG_SAVE, "Highlight self in kill messages")
```
Comment (heinrich5991) [inline src/game/variables.h:52]: It seems like this option is superfluous, you can set the kill message self highlight color to the normal one if you don't want them highlighted.

**Review by heinrich5991 — CHANGES_REQUESTED** [review]
Looks mostly good, thanks for the work.

In general, we try to minimize configuration options, but those for text colors are okay.

File: src/game/variables.h (hunk 1 above)
Comment (Azorlogh) [reply src/game/variables.h]: That's right, I removed it, thanks.

File: src/game/client/components/killmessages.cpp [hunk 2]
```diff
@@ -76,6 +76,15 @@ void CKillMessages::CreateKillmessageNamesIfNotCreated(CKillMsg &Kill)
 		TextRender()->SetCursor(&Cursor, 0, 0, FontSize, TEXTFLAG_RENDER);
 		Cursor.m_LineWidth = -1;
 
+		unsigned color = g_Config.m_ClKillMessageNormalColor;
```
[the remark this answers is not in the document]
Comment (Azorlogh) [reply src/game/client/components/killmessages.cpp]: Fixed

**Comment by heinrich5991** [comment]
bors r+

Thanks. :)

**Pull request merged**